calcom/cal.com
# fix: Teams/ Webhook Routes blocked based on features flag

## What does this PR do?


In this pr have made the changes , now webhook and teams route are accessible only if their respective flags are enabled. Previously even though teams was hidden on main sidebar when teams flag is disabled but when we go to settings, we can create or see teams from their and also the webhook even if it is disabled. Also a user can directly create team by going to /settings/teams/new  even though teams flag is disabled and also create webhooks even if webhook flag is disabled by the admin. 
Fixes 
Now with this Pull request , a user can be able to access teams and webhook routes only if they are enabled by the admin.
The settings sidebar will only show navigation items that are not disabled by the feature flags in the database. 
**### _Before making changes (Even when teams feature flag is off)_**
![image](https://github.com/beenthere-tech/cal.com/assets/66114276/332427c8-9920-49bc-9422-95440d42ef6f)
![image](https://github.com/beenthere-tech/cal.com/assets/66114276/d10dab65-5464-449d-848d-392c4cfd4a5c)


### **_After making changes and disabling teams feature flag_**

![image](https://github.com/beenthere-tech/cal.com/assets/66114276/2836b15e-96c7-4b6e-a6be-e4d627d995b6)
<img width="1318" alt="Screenshot 2023-09-11 at 10 33 12 PM" src="https://github.com/beenthere-tech/cal.com/assets/66114276/722ed8d1-ec57-42c0-914a-f5dd4b7a8f13">


### **_Settings Sidebar  when teams flag is disabled_** 

![image](https://github.com/calcom/cal.com/assets/66114276/e90fab8b-ac93-4bc9-a77d-72ff1844be3a)


### **_Settings Sidebar  when teams flag is enabled_** 
![image](https://github.com/beenthere-tech/cal.com/assets/66114276/d10dab65-5464-449d-848d-392c4cfd4a5c)

## Type of change


## How should this be tested?
1. Create an admin account.
2. Go to /settings/admin/flags
3. Disable or enable the teams / webhooks flag
4. See the results

## Mandatory Tasks
